File: software/examples/csharp/ExampleSimple.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XXYYZZ"; // Change XXYYZZ to the UID of your IMU Brick

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickIMU imu = new BrickIMU(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current quaternion
		float x, y, z, w;
		imu.GetQuaternion(out x, out y, out z, out w);

		Console.WriteLine("Quaternion[X]: " + x);
		Console.WriteLine("Quaternion[Y]: " + y);
		Console.WriteLine("Quaternion[Z]: " + z);
		Console.WriteLine("Quaternion[W]: " + w);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{imu_brick::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XXYYZZ"; // Change XXYYZZ to the UID of your IMU Brick.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let imu = ImuBrick::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current quaternion.
    let quaternion = imu.get_quaternion().recv()?;

    println!("Quaternion [X]: {}", quaternion.x);
    println!("Quaternion [Y]: {}", quaternion.y);
    println!("Quaternion [Z]: {}", quaternion.z);
    println!("Quaternion [W]: {}", quaternion.w);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
